### ISSUE TITLE
All Requests Return 401 from real servers

### ISSUE THREAD

**Issue by t8y8** [tableau/server-client-python#88]
The refactoring in #87 has a bug in our header handling.

```python
    @staticmethod
    def _make_common_headers(auth_token, content_type):
        retval = {}
        if auth_token is not None:
            retval['x-tableau-auth'] = auth_token
        if content_type is not None:
            retval['content-type'] = content_type
```

There's no return -- so the x-tableau-auth never gets updated, and all calls are missing the token.

I can whip up a patch.

/cc @RussTheAerialist

**Comment by t8y8** [comment]
Closed by #89

**Comment by RussTheAerialist** [comment]
I'm going to reopen this so that we can write a test to make sure something like this can never happen again.  We should have the habit of "if there is a bug, especially this critical, we should write a test to cover it" because it means there was a gap in our test coverage.

**Comment by t8y8** [comment]
Makes sense.

I looked into request-mocks docs a bit and they can verify headers and requests as well, I think 'real' tests are great but we can still get more coverage with the mocks

**Comment by LGraber** [comment]
Sounds good.

**Comment by t8y8** [comment]
I think I resolved this with #91.

We've got more enhancements coming in #110 as well -- can I close this @RussTheAerialist ?
